docs/examples/contact-manager.md

```diff
@@ -0,0 +1,215 @@
+
+The first thing we need to do is to import the `Skill` class and the `regex_matcher` from opsdroid. Then we will need to import `sqlite3` modules. Your `__init__.py` file should look like this:
+
+```python
+import sqlite3
+
+from opsdroid.matchers import match_regex
+from opsdroid.skill import Skill
+```
+
+### Connecting to SQLite3 
+You will create a python file with your database. After compiling it ```python mydatabase.py```, the ```mydatabase.db``` will be created.
+
+Our next step is to connect our `__init__.py` file to a SQLite database. Your `__init__.py` file should look something like this now:
+
+```python
+import sqlite3
+
+from opsdroid.matchers import match_regex
+from opsdroid.skill import Skill
+
+conn = sqlite3.connect('/path/to/my/database')
+c = conn.cursor()
+```
+
+#### Creating the Table
+
+We now have to use the SQLite3 database file that we had connected and add a table to it by doing 
+
+```python
+import sqlite3
+
+# Connect to SQLite database (creates the database file if it doesn't exist)
+connection = sqlite3.connect('sqllite.db')
+
+# Create a cursor object to execute SQL commands
+cursor = connection.cursor()
+
+# Create a table
+cursor.execute('''CREATE TABLE IF NOT EXISTS contacts (
+                    phoneNumber TEXT PRIMARY KEY,
+                    name TEXT NOT NULL,
+                    surname TEXT NOT NULL,
+                    email TEXT
+                )''')
+
+
+# Commit the transaction
+connection.commit()
+
+
+# Close cursor and connection
+cursor.close()
+connection.close()
+```
+
+With our table and database file we are set for the next section!
+
+#### Adding to the contact database
+
+Now we will be making a function which will be responsible for inserting the contact's data into the database. We will get the info from the database with `c.execute("SELECT * FROM contacts WHERE phoneNumber = ?", (phoneNumber,)")` and we will inserting it with `c.execute("INSERT INTO contacts VALUES (?)", (params,))`, params will equal contact's phone number, name, surname and email. Here is what it should look like:
+
+```python
+class Contact(Skill):
+ @match_regex(r"Add contact: (\d+), (.+), (.+), (\S+@\S+)") #phoneNumber, name, surname, email
+ async def addContact(self, message):
+  phoneNumber = message.regex.group(1)  # Extract the phoneNumber
+  name = message.regex.group(2) 
+  surname = message.regex.group(3) 
+  email =  message.regex.group(4) 
+  c.execute("SELECT * FROM contacts WHERE phoneNumber = ?", (phoneNumber,))
+  existing_contact = c.fetchone()
+  if existing_contact:
+    await message.respond(f"Contact already exists!")
+  else:
+    c.execute("INSERT INTO contacts (phoneNumber, name, surname, email) VALUES (?, ?, ?, ?)", (phoneNumber, name, surname, email))
+    conn.commit()
+  await message.respond(f"Contact with Phone Number {phoneNumber} added to the Contact Book!")
```

Comment:
I think you misunderstand, the spacing here is still incorrect. Each indentation needs to consist of four spaces, but all the code blocks here only use one space. Please can you fix this up.

Reply:
Okay, i see sorry, of course!